announcement/announcement-impl/impl/src/java/org/sakaiproject/announcement/impl/BaseAnnouncementService.java

```diff
@@ -202,7 +202,9 @@ public void init()
 			
 			TransformerFactory tFactory = TransformerFactory.newInstance();
 			docTransformer = tFactory.newTransformer();
-			
+
+			siteService.addSiteRemovalAdvisor(this);
```

Comment:
it probably would make sense to make sure you call
org.sakaiproject.site.impl.BaseSiteService#removeSiteRemovalAdvisor
in the destroy method

Reply:
Done